### PR TITLE
update mace to mace-torch and update apax

### PR DESCRIPTION
**Pull request by PythonFZ**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ quippy-ase = { version = "^0.9.12", optional = true }
 xmltodict = "^0.13.0"
 
 nequip = { version = "^0.5.6", optional = true }
-apax = { git = "https://github.com/apax-hub/apax.git", rev = "dev", optional = true}
```
Comment (M-R-Schaefer) [inline pyproject.toml:32]: should we wait until we have merged
https://github.com/apax-hub/apax/pull/229
on the apax side?

File: ipsuite/base/base.py [hunk 2]
```diff
@@ -50,17 +51,22 @@ def update_data(self):
...
             except FileNotFoundError:
                 # File can not be opened with DVCFileSystem, try normal open
-                return list(ase.io.iread(self.data_file))
+                data = list(ase.io.iread(self.data_file))
         else:
             raise ValueError("No data given.")
 
+        if self.data_id is not None:
+            return [data[self.data_id]]
+        else:
```
Comment (PythonFZ) [inline ipsuite/base/base.py:67]: Changes to this file should be in a dedicated PR

**Review by M-R-Schaefer — APPROVED** [review]
